raiden/network/transport/matrix/rtc/web_rtc.py

```diff
@@ -350,135 +372,131 @@ async def send_message(self, message: str) -> None:
+        self._stop_event = stop_event
+        # the addresses for which we're currently trying to initialize WebRTC channels
+        self._web_rtc_channel_inits: Set[Address] = set()
+        self._address_to_connection: Dict[Address, _RTCConnection] = {}
+        self.log = log.bind(node=to_checksum_address(node_address))
+
+    def _handle_message(self, message_data: str, partner_address: Address) -> None:
+        messages: List[ReceivedRaidenMessage] = []
+        for msg in validate_and_parse_message(message_data, partner_address):
+            messages.append(ReceivedRaidenMessage(message=msg, sender=partner_address))
+        self._process_messages(messages)
+
+    def _handle_ice_connection_closed(self, conn: _RTCConnection) -> None:
+        # the connecting may have already been removed by close_connection
+        self._address_to_connection.pop(conn.partner_address, None)
+        self.health_check(conn.partner_address)
 
     def _wrapped_initialize_web_rtc(self, address: Address) -> None:
-        self._web_rtc_channel_inits.add(address)
-        try:
-            return self._initialize_web_rtc(address)
-        finally:
-            self._web_rtc_channel_inits.remove(address)
+        if address in self._web_rtc_channel_inits:
+            return
+        attempt = 0
+        while attempt < 3 and not self.has_ready_channel(address):
+            self._web_rtc_channel_inits.add(address)
+            try:
+                self._initialize_web_rtc(address)
+            finally:
+                self._web_rtc_channel_inits.remove(address)
+            attempt += 1
 
     def _initialize_web_rtc(self, partner_address: Address) -> None:
         if self._stop_event.is_set():
             return
 
         self.log.debug(
-            "Initiating web rtc",
+            "Establishing WebRTC channel",
             partner_address=to_checksum_address(partner_address),
         )
 
-        rtc_partner = self.get_rtc_partner(partner_address)
+        conn = _RTCConnection(
+            partner_address,
+            self.node_address,
+            self._signaling_send,
+            self._handle_ice_connection_closed,
+            self._handle_message,
+        )
+        self._add_connection(partner_address, conn)
         self.schedule_task(
-            coroutine=rtc_partner.initialize_signalling(),
-            callback=self._handle_sdp_callback,
-            partner_address=partner_address,
+            coroutine=conn.initialize_signalling(),
+            callback=conn.handle_sdp_callback,
         )
 
-        # wait for WEB_RTC_CHANNEL_TIMEOUT seconds and check if connection was established
-        if self._stop_event.wait(timeout=WEB_RTC_CHANNEL_TIMEOUT):
+        # wait for _WEB_RTC_CHANNEL_TIMEOUT seconds and check if connection was established
+        if self._stop_event.wait(timeout=_WEB_RTC_CHANNEL_TIMEOUT):
             return
 
         # if room is not None that means we are at least in the second iteration
```

Comment:
This is legacy comment, could you remove that?

Reply:
Done.